bitwarden_license/test/Scim.Test/Groups/GetGroupsListQueryTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Repositories;
+using Bit.Scim.Groups;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using Bit.Test.Common.Helpers;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Scim.Test.Groups;
+
+[SutProviderCustomize]
+public class GetGroupsListCommandTests
+{
+    [Theory]
+    [BitAutoData(10, 1)]
+    [BitAutoData(2, 1)]
+    [BitAutoData(1, 3)]
+    public async Task GetGroupsList_Success(int count, int startIndex, SutProvider<GetGroupsListQuery> sutProvider, Guid organizationId, IList<Group> groups)
+    {
+        groups = SetGroupsOrganizationId(groups, organizationId);
+
+        sutProvider.GetDependency<IGroupRepository>()
+            .GetManyByOrganizationIdAsync(organizationId)
+            .Returns(groups);
+
+        var result = await sutProvider.Sut.GetGroupsListAsync(organizationId, null, count, startIndex);
+
+        await sutProvider.GetDependency<IGroupRepository>().Received(1).GetManyByOrganizationIdAsync(organizationId);
```

Comment:
I wouldn't assert this call, we shouldn't care where the query gets its groups from, only that it returns the correct result (which we're testing below). Same for the other tests in this PR.

To explain further: I mostly say this because it's a query. The caller only cares that it gets the correct data back. Whereas with a command, we want to make sure it does actually update the database, so asserting the repository call is more important. In both cases the idea is to focus on outcomes rather than implementation details, although that's not always possible and there's definitely room for disagreement.

Reply:
That makes very much sense to me! Updated the branch with your suggestion.